config/http_config.go

```diff
@@ -113,24 +113,24 @@ func newClient(rt http.RoundTripper) *http.Client {
 
 // NewClientFromConfig returns a new HTTP client configured for the
 // given config.HTTPClientConfig. The name is used as go-conntrack metric label.
-func NewClientFromConfig(cfg HTTPClientConfig, name string) (*http.Client, error) {
-	rt, err := NewRoundTripperFromConfig(cfg, name)
+func NewClientFromConfig(cfg HTTPClientConfig, name string) (*http.Client, func(), error) {
```

Comment:
I was thinking rather than changing the return values, to instead have an interface that includes http.Client and also the new method. That way it's there for those who need it, and doesn't pollute the API usage for others.